core/rawdb/accessors_da_syncer.go

```diff
@@ -0,0 +1,39 @@
+package rawdb
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/ethdb"
+	"github.com/scroll-tech/go-ethereum/log"
+)
+
+// WriteDASyncedL1BlockNumber writes the highest synced L1 block number to the database.
+func WriteDASyncedL1BlockNumber(db ethdb.KeyValueWriter, L1BlockNumber uint64) {
+	value := big.NewInt(0).SetUint64(L1BlockNumber).Bytes()
+
+	if err := db.Put(daSyncedL1BlockNumberKey, value); err != nil {
+		log.Crit("Failed to update da synced L1 block number", "err", err)
+	}
+}
+
+// ReadDASyncedL1BlockNumber retrieves the highest synced L1 block number.
+func ReadDASyncedL1BlockNumber(db ethdb.Reader) *uint64 {
+	data, err := db.Get(daSyncedL1BlockNumberKey)
+	if err != nil && isNotFoundErr(err) {
+		return nil
+	}
+	if err != nil {
+		log.Crit("Failed to read da synced L1 block number from database", "err", err)
+	}
```

Comment:
return nil in this block, there is no guarantee that data is nil in case of error.

Reply:
don't get it

Reply:
or perhaps better, change the interface to return errors.

Reply:
the issue is that there is an error and you are proceeding below like there was no error.

Either we should return the error from the interface, or we should short circuit; we can't proceeed.

Reply:
but `log.Crit` is not just logging, it is also terminates program

Reply:
oh, i see.

Fine in that regard.